### PR TITLE
fix(Message): update typings and docs related to #edit

### PR DESCRIPTION
**Pull request by iShibi**
**Please describe the changes this PR makes and why it should be merged:**

This PR fixes the following:
1. The jsdoc of `Message#edit` doesn't show `content` can be null
2. `Message#edit` can take `MessageAttachment` or array of it in `options`
3. The `content` field in `MesssageEditOptions` can be null
4. The jsdoc of `MessageEditOptions` doesn't show `content` can be null

Thanks X 3, @ToasterSticks for noticing this issue too 👍 

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Review by SpaceEEC — APPROVED** [review]

File: src/structures/Message.js [hunk 1]
```diff
@@ -474,7 +474,7 @@ class Message extends Base {
   /**
    * Options that can be passed into {@link Message#edit}.
    * @typedef {Object} MessageEditOptions
-   * @property {string} [content] Content to be edited
+   * @property {string|null} [content] Content to be edited
```
Comment (vaporox) [inline src/structures/Message.js:477]:
```suggestion
   * @property {?string} [content] Content to be edited
```

File: src/structures/Message.js [hunk 2]
```diff
@@ -486,8 +486,8 @@ class Message extends Base {
 
   /**
    * Edits the content of the message.
-   * @param {string|APIMessage} [content] The new content for the message
-   * @param {MessageEditOptions|MessageEmbed} [options] The options to provide
+   * @param {string|APIMessage|null} [content] The new content for the message
```
Comment (vaporox) [inline src/structures/Message.js:489]:
```suggestion
   * @param {?string|APIMessage} [content] The new content for the message
```

**Review by vaporox — COMMENTED** [review]
In what case would you want to do `message.edit(null);` though?

File: src/structures/Message.js (hunk 1 above)
Comment (iShibi) [reply src/structures/Message.js:477]: When you want to send an attachment or embed in the edit but want to remove the `content`

**Review by kyranet — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**